config/routes.rb

```diff
@@ -33,6 +33,14 @@
     resources :messages, only: %i[index new create destroy] do
       get :reply, on: :collection
     end
+
+    collection do
+      get '/nbp_wallet/connect', to: 'users/nbp_wallet#connect'
+      get '/nbp_wallet/qr_code', to: 'users/nbp_wallet#qr_code'
+      post '/nbp_wallet/relationship_status', to: 'users/nbp_wallet#relationship_status'
+      get '/nbp_wallet/finalize', to: 'users/nbp_wallet#finalize'
```

Comment:
Normally, I would tend to switch the HTTP verbs between both actions (it's rather about "getting" the current status, isn't it?), but redirecting with a POST doesn't work (one would need to submit a form, which is not worth the effort -- GET is fine).